pkg/cdi/spec_unix.go

```diff
@@ -0,0 +1,51 @@
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris zos
+
+/*
+   Copyright © 2021 The CDI Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package cdi
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+// Rename src in dir to dst in the same directory. If dst already exists
+// refuse renaming with an error unless overwrite is explicitly asked for.
+func rename(dir, src, dst string, overwrite bool) error {
```

Comment:
Question: Is it strictly necessary that these be in the same directory? If so, should this be `renameIn` or something?

Reply:
Not in the native syscall (`renameat2()`). It's just the way how we use it as an unexported/internal function. Maybe we should just call it `renameat()`. That's how the syscall would be called in an ideal world if the API would have been gotten right at the first try.

Reply:
I'd suggest to call it renameAt2 as renameAt could make people think that it uses renameat syscall.